addons/mrp/models/mrp_production.py

```diff
@@ -595,7 +595,8 @@ def _onchange_product_qty(self):
     @api.onchange('bom_id')
     def _onchange_bom_id(self):
         if not self.product_id and self.bom_id:
-            self.product_id = self.bom_id.product_id or self.bom_id.product_tmpl_id.product_variant_ids[0]
+            self.product_id = self.bom_id.product_id \
+                                or self.bom_id.product_tmpl_id.product_variant_ids[0] if self.bom_id.product_tmpl_id.product_variant_ids else False
```

Comment:
It isn't just the same than doing that ?
```suggestion
            self.product_id = self.bom_id.product_id or self.bom_id.product_tmpl_id.product_variant_ids[:1]
```

Reply:
Yes, thank you!